File: 00. Estructura/01. Lenguajes de proposito general/02. Programación básica/08. Estructuras/02. Ciclos/03. For/Java/For.java
package prueba;

public class Prueba {
    
    public static void main(String[] args) {
       
       // Estandar 
        for (definición; condición; asignación) {
            // Contenido
        	continue;		// Saltar a la verificación de condición
            // Contenido	
        	break;			// Salir del ciclo
            // Contenido
        }

        // Con retorno
        ArrayList list = new ArrayList(); 
        for (definición; condición; asignación) {
        	list.add(expresión);
        }
        Object[] retorno = list.toArray();

        // Extendido (Recorre los elemenentos de un arreglo o un objeto iterable)
        for (tipo elemento : arreglo_o_iterable) {
            // Contenido
        }

        // Extendido con retorno (Recorre los elemenentos de un arreglo o un objeto iterable)
        ArrayList list = new ArrayList();               // Extendido
        for (tipo elemento : arreglo_o_iterable) {
            list.add(expresión);
        }
        Object[] retorno = list.toArray();
        ArrayList list = new ArrayList();               // Extendido con if
        for (tipo elemento : arreglo_o_iterable) {
            if (condición) 
                list.add(expresión);
        }
        Object[] retorno = list.toArray();
        ArrayList list = new ArrayList();               // Extendido con if - else
        for (tipo elemento : arreglo_o_iterable) {
            if (condición) 
                list.add(expVerdadera);
            else 
                list.add(expFalsa);
        }
        Object[] retorno = list.toArray();

    }

}

File: 00. Estructura/01. Lenguajes de proposito general/02. Programación básica/08. Estructuras/02. Ciclos/03. For/JavaScript/For.js
  
// Estandar 
for (definición; condición; asignación) {
    // Contenido
	continue;		// Saltar a la verificación de condición
    // Contenido	
	break;			// Salir del ciclo
    // Contenido
}

// Con retorno
retorno = [] 
for (definición; condición; asignación) {
	retorno.push(expresión);
}

// Extendido 1 (Recorre los elemenentos de un arreglo o un objeto iterable)
for (e of arreglo_o_iterable) {
    // Contenido
}

// Extendido 2 (Recorre las posiciones o indicadores de un arreglo o un objeto iterable)
for (i in arreglo_o_iterable) {
    // Contenido
}

// Extendido con retorno (Recorre los elemenentos de un arreglo o un objeto iterable)
retorno = arreglo_o_iterable.map(e => expresión);                                                         // Extendido
retorno = arreglo_o_iterable.map(e => condición ? expresión : undefined).filter(i => i !== undefined);    // Extendido con if
retorno = arreglo_o_iterable.map(e => condición ? expVerdadera : expFalsa);                               // Extendido con if - else
